core/src/main/java/org/elasticsearch/index/query/QueryStringQueryParser.java

```diff
@@ -164,6 +164,8 @@ public QueryBuilder fromXContent(QueryParseContext parseContext) throws IOExcept
                 } else {
                     throw new ParsingException(parser.getTokenLocation(), "[query_string] query does not support [" + currentFieldName + "]");
                 }
+            } else {
+                throw new ParsingException(parser.getTokenLocation(), "[query_string] unknown token [" + token + "] after [" + currentFieldName + "]");
```

Comment:
You switched to using the query builder NAME in other places - why not here as well?